libsolidity/codegen/LValue.cpp

```diff
@@ -155,9 +155,14 @@ ImmutableItem::ImmutableItem(CompilerContext& _compilerContext, VariableDeclarat
 void ImmutableItem::retrieveValue(SourceLocation const&, bool) const
 {
 	solUnimplementedAssert(m_dataType->isValueType(), "");
-	solAssert(!m_context.runtimeContext(), "Tried to read immutable at construction time.");
-	for (auto&& slotName: m_context.immutableVariableSlotNames(m_variable))
-		m_context.appendImmutable(slotName);
+
+	if (m_context.runtimeContext())
+		CompilerUtils(m_context).loadFromMemory(
+			static_cast<unsigned>(m_context.immutableMemoryOffset(m_variable))
+		);
```

Comment:
You definitely have to pass the type to this in general. (But that shouldn't affect the test case I think you just said is throwing)